lib/Service/FederationService.php

```diff
@@ -73,6 +73,14 @@ public function getRemoteCollaboraURL($remote) {
 		if (!$this->isTrustedRemote($remote)) {
 			throw new \Exception('Unable to determine collabora URL of remote server ' . $remote . ' - Remote is not a trusted server');
 		}
+
+		$serverProtocol = $this->request->getServerProtocol() . '://';
+		$serverHost = $this->request->getServerHost();
+		$server = $serverProtocol . $serverHost . $this->request->getRequestUri();
+		if (str_starts_with($server, $remote)) {
```

Comment:
We can actually use https://github.com/nextcloud/server/blob/dae7c159f728a90ffa53247d6e033abdae5d2bd6/lib/public/Security/ITrustedDomainHelper.php#L27 to check that which then would also cover shares on the same instance with different urls